Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -2413,7 +2413,15 @@ eMovementState CClientPed::GetMovementState()
             if (cs.LeftStickX == 0 && cs.LeftStickY == 0)
                 return MOVEMENTSTATE_CROUCH;
             else
+            {
+                // Is he aiming and rolling while crouching?
+                if ((cs.LeftStickX == -128 || cs.LeftStickX == 128) && cs.RightShoulder1)
+                    return MOVEMENTSTATE_ROLL;
```

Comment:
Btw, how did you figure out this if-check?

Reply:
https://github.com/multitheftauto/mtasa-blue/blob/927509565ab3a8d48df006440654c38e0942aa89/Client/sdk/game/CPad.h#L20
https://github.com/multitheftauto/mtasa-blue/blob/927509565ab3a8d48df006440654c38e0942aa89/Client/sdk/game/CPad.h#L27

Reply:
@qaisjp How about this?

```
bool CControllerState::IsMoving()
{
    return cs.LeftStickX == -128 || cs.LeftStickX == 128;
}

bool CControllerState::IsAiming()
{
    return cs.RightShoulder1 != false;
}
```

```
if (cs.IsMoving() && cs.IsAiming())
    return MOVEMENTSTATE_ROLL;
```

Reply:
That makes me very happy

Reply:
make us*, thnx saml1er